### ISSUE TITLE
device credentials

### ISSUE THREAD

**Issue by elpinjo** [elpinjo/CumulocityClient#1]
I am working on registering a device on cumulocity using ESP32. I am able to connect, register a device and can send data. However, once the device is disconnected, I am unable to reconnect and have to end up creating a new device. I have seen the example in Python where before creating a device, the code checks if the device already exists or not. However, the example in Arduino using different code structure. Can you please help or share some Arduino code that connects and receive device credentials?

_Originally posted by @zabutt in https://github.com/arduino/Arduino/issues/9987#issuecomment-716785700_

**Comment by elpinjo** [comment]
@zabutt, you would need to store the credentials received during the device registration. I was already planning on extending the example to make use of the ESP32 SPIFFS filesystem.
I will work on that soon

**Comment by zabutt** [comment]
That would be a great help elpinjo.

**Comment by elpinjo** [comment]
I have created a new version of the library. Be aware I did change the API a bit to try and simplify it. 
The example that uses the ESP's SPIFFS file system to store device credentials is located in the example RetrieveAndStoreCredentials.

Note that there are two ways to connect your device to Cumulocity:
- Using static credentials which you create in the Cumulocity user interface. Go to Administration and create a user with the device role
- Using device registration. With device registration one needs the bootstrap credentials for Cumulocity. These can be obtained via Software AG Support. (or send me a direct message). With these bootstrap credentials one can request device credentials which need to be stored on the device itself.
